### DISCUSSION TITLE
KafkaUpgradeException

### DISCUSSION THREAD

**Discussion by vutkin** [strimzi/strimzi-kafka-operator#5030] [Q&A]
Hi, I've got a problem with upgrading operator from 0.19.0->0.20.1

 Message:               Unsupported Kafka.spec.kafka.version: 2.4.1. Supported versions are: [2.5.1, 2.6.0, 2.5.0]
    Reason:                KafkaUpgradeException

0.20.1 drops supports 2.4.1, unfortunately I've not upgraded it till supported versions.
I can't downgrade strimzi-operator due to helm2/3 bug with crd.
Is any way to upgrade kafka cluster?

**Comment by scholzj** [comment]
Well, you can try to change the version number to some supported version in the Kafka CR and see if it helps. I'm not sure what changes were between 0.19 and 0.20 anymore ... it might upgrade smoothly, but if it does not, just delete the Kafka pods / stateful set (without the persistent volumes) and the operator should recover it. 

In 0.22 we did some changes and this should be much easier if you would want to upgrade directly to 0.22 (but there is a CRD upgrade between 0.22 and 0.23, so it would be better to not go directly from 0.19 to 0.23, but first to 0.22 if you decide to go this way).

**Reply by vutkin** [reply]
Hi @scholzj, will it upgrade message version in this case without any problems?

**Reply by scholzj** [reply]
Do you have it set in the Kafka CR? If not, the 0.22 will do it step by step. 0.20 would just bump everything at once.

**Reply by vutkin** [reply]
Yes I have, look here please
```
kafka:
  config:
    log.message.format.version: "2.5"
  version: 2.5.0
```

**Reply by scholzj** [reply]
Where you are starting from is more important. The old Kafka CR had Kafka version 2.4.1. Did it also had `log.message.format.version: 2.4`? If so, change only the `version` field at first. You can change `log.message.format.version` filed later after the upgrade.

**Reply by vutkin** [reply]
Yes, it has `log.message.format.version: 2.4`. My question is: could I set the version to 2.5 and at same time set `log.message.format.version: 2.5`? and then do
```
k delete sts test-kafka -n kafka --cascade=orphan
```
Thank you for advises.

**Reply by scholzj** [reply]
If you delete the STS then it probably doesn't matter. But normally you should change first the version and only afterwards thee `log.message.format.version`. If you had it before set to 2.4 I think the upgrade might go fine if you just change the `version` field.
